src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -147,14 +147,16 @@ runLanguageServer options userHandlers onInitialConfig onConfigChange getIdeStat
                               Left e  -> sendFunc $ wrap $ ResponseMessage "2.0" (responseId _id) Nothing (Just e)
                               Right r -> sendFunc $ wrap $ ResponseMessage "2.0" (responseId _id) (Just r) Nothing
                     ResponseAndRequest x@RequestMessage{_id, _params} wrap wrapNewReq act ->
-                        checkCancelled ide clearReqId waitForCancel lspFuncs wrap act x _id _params $
+                        checkCancelled ide clearReqId waitForCancel lspFuncs wrap act x _id _params $ do
                             \(res, newReq) -> do
-                            sendFunc $ wrap $ ResponseMessage "2.0" (responseId _id) (Just res) Nothing
-                            case newReq of
-                                Nothing -> return ()
-                                Just (rm, newReqParams) -> do
-                                    reqId <- getNextReqId
-                                    sendFunc $ wrapNewReq $ RequestMessage "2.0" reqId rm newReqParams
+                                case res of
+                                    Left e  -> sendFunc $ wrap $ ResponseMessage "2.0" (responseId _id) Nothing (Just e)
+                                    Right r -> sendFunc $ wrap $ ResponseMessage "2.0" (responseId _id) (Just r) Nothing
+                                case newReq of
```

Comment:
I'd have written: `whenJust newReq $ \(rm, newReqParams) -> ...` - but that's how I'd have written the previous one, so personal preference.

Reply:
Definitely in favor of using `whenJust`.